modules/extensions-support/src/main/java/org/mule/module/extension/internal/runtime/DefaultExecutionMediator.java

```diff
@@ -206,4 +245,23 @@ private void intercept(List<Interceptor> interceptors, Consumer<Interceptor> clo
         });
     }
 
+    private Throwable handleException(Exception e)
+    {
+        Optional<Throwable> connectionException = Arrays.stream(getThrowables(e)).filter(t -> t instanceof ConnectionException).findFirst();
```

Comment:
THis doesn't seem right. Modify ExceptionUtils so that getRootCause() stops the recursion at a ConnectionException. Otherwise we would have to repeat logic like this in many places